### PR TITLE
Set /bin/bash to bash shell in docker command.

### LINKED ISSUE #1704: Prod worker can not be started due to "exec: \"bash\": executable file not found in $PATH": unknown"
```
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/codalab/worker/worker.py", line 56, in start
    self._run_manager.process_runs()
  File "/opt/codalab/worker/local_run/local_run_manager.py", line 190, in process_runs
    self._runs[bundle_uuid] = self._run_state_manager.transition(run_state)
  File "/opt/codalab/worker/fsm.py", line 35, in transition
    return self._transition_functions[state.stage](state)
  File "/opt/codalab/worker/local_run/local_run_state.py", line 292, in _transition_from_PREPARING
    runtime=self.docker_runtime,
  File "/opt/codalab/worker/docker_utils.py", line 34, in wrapper
    raise DockerException(message + ': ' + str(e))
codalab.worker.docker_utils.DockerException: Unable to start Docker container: 409 Client Error: Conflict ("Conflict. The container name "/codalab_run_0xc2559ab6e42f446b9beea50c74c1222f" is already in use by container "3d823a5f30232fb833b193da2d8839d380ac8c16114b60f0cc8f007caf4d10f7". You have to remove (or rename) that container to be able to reuse that name.")
2019-11-21 02:20:18,429 Sleeping for 1 hour due to exception...please help me!
```
This issue might need some investigation as I saw it several times in production.

### PR DESCRIPTION
**Pull request by candicegjing**
Fixed #1704. 
This PR will fix the problem caused by the following error: 
```"exec: \"bash\": executable file not found in $PATH": unknown"```

This error had periodically happened in production, shown as follows:
```
docker.errors.APIError: 400 Client Error: Bad Request ("OCI runtime create failed: container_linux.go:345: starting container process caused "exec: \"bash\": executable file not found in $PATH": unknown")
```
I had done some investigation, findings are all documented at the original ticket. Currently, PATH is set to have the following paths:
```
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```
Explicitly setting the path of bash shell in environment variable $PATH will help the program to navigate the right bash shell easier.

### CODE REVIEW

File: docker/dockerfiles/Dockerfile.worker [hunk 1]
```diff
@@ -11,6 +11,7 @@ FROM ubuntu:16.04
 MAINTAINER CodaLab Worksheets <codalab.worksheets@gmail.com>
 
 ENV DEBIAN_FRONTEND noninteractive
+ENV PATH="/bin/bash:${PATH}"
```
Comment (percyliang) [inline docker/dockerfiles/Dockerfile.worker]: Why is this necessary, especially since you have `/bin/bash` in the docker command above?

**Comment by percyliang** [comment]
How come this wasn't an issue before / when does this error occur?

File: docker/dockerfiles/Dockerfile.worker (hunk 1 above)
Comment (candicegjing) [reply docker/dockerfiles/Dockerfile.worker]: Theoretically yes. However, the existing `bash` is already linked to `/bin/bash`, but the program still can't find it when executing. I did several ways of testing, they are documented at [here](https://github.com/codalab/codalab-worksheets/issues/1704#issuecomment-575926090). It appears that setting only the `PATH` in the environment variable still can't help the program to navigate to the right executable. So I feel the safest way is to add BASH executable directly to `PATH` and call `/bin/bash` in the code.

**Comment by candicegjing** [comment]
> How come this wasn't an issue before / when does this error occur?

This issue actually happened in the past quarter when we had an error about **duplicate container name**. The **duplicate container name**  exception is triggered by **the unknown bash executable exception**. There seem to be many discussions about this issue. Some posts mentioned this is related to the specific version of the docker engine version: `18.09.*`, which is the same one we are using. Some of the properties within the docker config shouldn't be set (e.g. `MountFlags=slave`). But this is not our issue. We don't have that configuration in our docker service config file. Then another possible reason is from the command itself. The command may evoke a new environment that doesn't have the linked path in the original environment (e.g. subshell).

**Review by percyliang — COMMENTED** [review]
Could you please summarize the consequences of this change and any outstanding points that someone in the future can address?  Thanks!

**Comment by candicegjing** [comment]
> Could you please summarize the consequences of this change and any outstanding points that someone in the future can address? Thanks!

I added a comment above the line that declares the `/bin/bash`.  I think explicitly specifying this won't affect anything. The only thing could be bad is when `/bin/bash` doesn't exist in the system. But this should happen rarely.
